src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -69,4 +60,27 @@ private SwiftDescriptions() {
     output.enableObjcInterop = Optional.of(true);
   }
 
+  static boolean hasSwiftSource(ImmutableSortedSet<SourceWithFlags> srcs) {
+    return Iterables.any(srcs, new Predicate<SourceWithFlags>() {
+      @Override
+      public boolean apply(SourceWithFlags input) {
+        SourcePath sourcePath = input.getSourcePath();
+        return (sourcePath instanceof PathSourcePath) &&
```

Comment:
I think the approach of always having a fake swift placeholder rule, and just having it do nothing if there are no swift sources, which you used in the previous commit was good.